compiler-core/src/ast.rs

```diff
@@ -413,6 +413,47 @@ impl TypeAst {
             TypeAst::Var(_) | TypeAst::Hole(_) => Some(Located::Annotation(self.location(), type_)),
         }
     }
+
+    /// Consumes a `TypeAst` to return the corresponding annotation.
+    pub fn to_annotation(self) -> EcoString {
```

Comment:
This doesn't need to take ownership, let's have it take a reference.

Could we call it `print` too please, to match more what we have in other parts of the codebase.

Ideally this would call an internal helper function and append to a mutable EcoString rather than making lots of Strings